templates/single-product/related.php

```diff
@@ -46,9 +46,9 @@
 
 if ( $products->have_posts() ) : ?>
 
-	<div class="related products">
+	<section class="related products">
 
-		<h2><?php _e( 'Related Products', 'woocommerce' ); ?></h2>
+		<h2><?php echo sprintf( __( 'Other %s...', 'woocommerce' ), $product->get_categories() ); ?></h2>
```

Comment:
This return a list of categories: https://codex.wordpress.org/Function_Reference/get_the_term_list
And it's very common people use like 5 or 10 categories in a product...